src/kubeobjects/secret.go

```diff
@@ -70,6 +84,42 @@ func (query SecretQuery) CreateOrUpdate(secret corev1.Secret) error {
 	return nil
 }
 
+func (query SecretQuery) CreateOrUpdateForNamespacesList(newSecret corev1.Secret, namespaces []corev1.Namespace) (int, error) {
+	secretList, err := query.GetAllFromNamespaces(newSecret.Name)
+	if err != nil {
+		return 0, err
+	}
+
+	namespacesContainingSecret := make(map[string]corev1.Secret, len(secretList))
+	for _, secret := range secretList {
+		namespacesContainingSecret[secret.Namespace] = secret
+	}
+
+	updateAndCreationCount := 0
+
+	for _, namespace := range namespaces {
+		newSecret.Namespace = namespace.Name
+
+		if oldSecret, ok := namespacesContainingSecret[namespace.Name]; ok {
+			if !AreSecretsEqual(oldSecret, newSecret) {
+				err = query.Update(newSecret)
+				if err != nil {
+					return updateAndCreationCount, err
+				}
+				updateAndCreationCount++
+			}
+		} else {
+			err = query.Create(newSecret)
+			if err != nil {
+				return updateAndCreationCount, err
+			}
+			updateAndCreationCount++
+		}
+	}
+
```

Comment:
what would be kinda cool, is that you could delete the leftover/not-necessary secrets. (stuff in the `namespacesContainingSecret` that were not looked at during the previous loop)
but then the functions name is not that accurate 😅

Reply:
This is done per dynakube, what if there are 2 application monitoring dynakubes on different namespaces? - One reconciliation would then delete the secrets of the other, wouldn't it?

Reply:
If the name of the secret is unique to a dynakube then no,
if the name of the secret is always the same then yes, (and I think this is the case, its always `dynatrace-dynakube-config`)

so its a possible future improvement 😉